webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -59,12 +59,16 @@ export class ContactsEffects {
 
         const loadContact = this
           .loadContact(id)
-          .then(() => this.loadChildren(userFacilityId))
-          .then(() => this.loadReports(forms))
-          .then(() => this.loadTargetDoc())
-          .then(() => this.loadContactSummary())
-          .then(() => this.loadTasks())
+          .then(() => this.loadChildren(id, userFacilityId))
+          .then(() => this.loadReports(id, forms))
+          .then(() => this.loadTargetDoc(id))
+          .then(() => this.loadContactSummary(id))
+          .then(() => this.loadTasks(id))
           .catch(err => {
+            // If the selected contact changed, just stop loading this one
+            if (err.code === 409) {
```

Comment:
@njogz @dianabarsan I am not sure about the best way in Javascript to qualify on which error was actually caught. I see other places in the code were we do check the HTTP status in `err.code`, so I went with that here. However, this does not seem super precise (especially in such a long/nested chain of promises) and `409` also seems kind of a strange status to have when not doing a POST.

I also considered just using a boolean field on `err`:

```.js
            if (err.selectedContactChanged) {
              return of();
            }
```

Or I could repeat the validation check here:
```.js
            if (this.selectedContact?._id !== id) {
              return of();
            }
```

Reply:
May I ask why you're checking for a 409? OH! 
Are you by any chance seeing something about `_local/rulesStateStore` in the logs when this error is thrown?

Reply:
The `409` is coming from [my code](https://github.com/medic/cht-core/pull/7159/files#diff-13fc702be7e06b5df657cee556e29c679f8ea364ae63ce6a79d96637881f93bdR104) on line 104.  That is just what I am using to match up the error I throw in `verifySelectedContactNotChanged` with this logic here in the `catch`.  Like I said, though, this does not feel like the right way to do this....

Reply:
It doesn't :) 409 means something specific in PouchDB/CouchDB world and this creates massive confusion. I'd just use `SOME_STRING_LITERAL_ERROR_CODE` that is human readable and suggestive.